extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-preview.js

```diff
@@ -0,0 +1,654 @@
+      const scaleWidthChange = deviceSpecs.width * (scale - 1) * 0.5; // Accounts for width change on scaling
+      const cumPaddings = (i + 1) * paddingSize;
+      const leftOffset = cumPaddings + cumWidthSum + scaleWidthChange;
+      setStyles(deviceSpecs.element, {
+        'transform': `perspective(100px) translateZ(${
+          (100 * (scale - 1)) / scale
+        }px) translateX(${leftOffset / scale}px)`,
+      });
+      cumWidthSum += deviceSpecs.width * scale;
+    });
+  }
+
+  /**
+   * @private
+   */
+  showAddDevicePopup_() {
+    const popup = buildAddDevicePopupTemplate(this.element);
+    popup
+      .querySelector('.i-amphtml-story-dev-tools-device-popup-close')
+      .addEventListener('click', () => {
+        setTimeout(() => {
+          popup.remove();
+        }, 200);
+        popup.removeAttribute('active');
+      });
+    const sections = ['mobile', 'tablet', 'desktop'].reduce((obj, section) => {
+      obj[section] = popup.querySelector(
+        `.i-amphtml-story-dev-tools-device-popup-${escapeCssSelectorIdent(
+          section
+        )}`
+      );
+      return obj;
+    }, {});
+    ALL_SCREEN_SIZES.forEach((device) => {
+      const deviceChip = buildDeviceChipTemplate(this.element);
+      const correspondingDevice = this.devices_.find(
+        (d) => d.name == device.name
+      );
+      deviceChip.device = correspondingDevice || device;
+      if (!correspondingDevice) {
+        deviceChip.setAttribute('inactive', '');
+      }
+      deviceChip.querySelector('span').textContent = device.name;
+      let chipSection = sections['mobile'];
+      if (device.width / device.height > 1) {
+        chipSection = sections['desktop'];
+      } else if (device.width / device.height > 0.75) {
+        chipSection = sections['tablet'];
+      }
+      chipSection.appendChild(deviceChip);
+    });
+
+    popup.addEventListener('click', (event) => {
+      const chip = closest(event.target, (el) => el.device);
+      if (!chip) {
+        return;
+      }
+      const deviceWasActive = chip && !chip.hasAttribute('inactive');
+      if (deviceWasActive) {
+        this.removeDevice_(chip.device);
+      } else {
+        this.addDevice_(chip.device);
+      }
+      chip.toggleAttribute('inactive', deviceWasActive);
+      this.updateDevicesInHash_();
+    });
+
+    // Add popup to screen.
+    this.element.appendChild(popup);
+    setTimeout(() => {
+      popup.setAttribute('active', '');
+    }, 1);
+  }
+
+  /**
+   * @private
+   */
+  showHelpPopup_() {
+    const popup = buildHelpPopupTemplate(this.element);
+    popup
+      .querySelector('.i-amphtml-story-dev-tools-device-popup-close')
+      .addEventListener('click', () => {
+        setTimeout(() => {
+          popup.remove();
+        }, 200);
+        popup.removeAttribute('active');
+      });
+
+    popup.querySelector(
+      '.i-amphtml-story-dev-tools-help-search-preview-link'
+    ).href += this.storyUrl_;
```

Comment:
The search preview link requires the `url` queryParam so we add the storyUrl to it